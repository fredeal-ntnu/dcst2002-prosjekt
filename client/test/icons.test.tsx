import React from 'react';
import { shallow } from 'enzyme';
import {
  EyeIcon,
  SearchIcon,
  QuestionIcon,
  FavouriteIcon,
  UpvoteIcon,
  DownvoteIcon,
  CommentBubleIcon,
} from 'src/icons';

describe('EyeIcon', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<EyeIcon size={24} color="blue" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('SearchIcon', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SearchIcon size={20} color="red" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('QuestionIcon', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<QuestionIcon size={24} color="green" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('FavouriteIcon', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<FavouriteIcon size={24} color="yellow" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('UpvoteIcon', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<UpvoteIcon size={24} color="green" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('DownvoteIcon', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<DownvoteIcon size={24} color="red" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('CommentBubleIcon', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CommentBubleIcon size={24} color="blue" />);
    expect(wrapper).toMatchSnapshot();
  });
});
