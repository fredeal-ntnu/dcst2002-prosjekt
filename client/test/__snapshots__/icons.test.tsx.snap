// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBubleIcon renders correctly 1`] = `
<svg
  dangerouslySetInnerHTML={
    {
      "__html": "
    <path d="M16 8c0 3.866-3.582 7-8 7a9.06 9.06 0 0 1-2.347-.306c-.584.296-1.925.864-4.181 1.234-.2.032-.352-.176-.273-.362.354-.836.674-1.95.77-2.966C.744 11.37 0 9.76 0 8c0-3.866 3.582-7 8-7s8 3.134 8 7zM5 8a1 1 0 1 0-2 0 1 1 0 0 0 2 0zm4 0a1 1 0 1 0-2 0 1 1 0 0 0 2 0zm3 1a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"/>
    ",
    }
  }
  fill="blue"
  height={24}
  style={
    {
      "verticalAlign": "-0.125em",
    }
  }
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`DownvoteIcon renders correctly 1`] = `
<svg
  dangerouslySetInnerHTML={
    {
      "__html": "
<path d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm6.5 4.5v5.793l2.146-2.147a.5.5 0 0 1 .708.708l-3 3a.5.5 0 0 1-.708 0l-3-3a.5.5 0 1 1 .708-.708L7.5 10.293V4.5a.5.5 0 0 1 1 0z"/>",
    }
  }
  fill="red"
  height={24}
  style={
    {
      "verticalAlign": "-0.125em",
    }
  }
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`EyeIcon renders correctly 1`] = `
<svg
  dangerouslySetInnerHTML={
    {
      "__html": "
      <path d="M16 8s-3-5.5-8-5.5S0 8 0 8s3 5.5 8 5.5S16 8 16 8zM1.173 8a13.133 13.133 0 0 1 1.66-2.043C4.12 4.668 5.88 3.5 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.133 13.133 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755C11.879 11.332 10.119 12.5 8 12.5c-2.12 0-3.879-1.168-5.168-2.457A13.134 13.134 0 0 1 1.172 8z"/>
      <path d="M8 5.5a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5zM4.5 8a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0z"/>
    ",
    }
  }
  fill="blue"
  height={24}
  style={
    {
      "verticalAlign": "-0.125em",
    }
  }
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`FavouriteIcon renders correctly 1`] = `
<svg
  dangerouslySetInnerHTML={
    {
      "__html": "
  <path d="M2 15.5a.5.5 0 0 0 .74.439L8 13.069l5.26 2.87A.5.5 0 0 0 14 15.5V2a2 2 0 0 0-2-2H4a2 2 0 0 0-2 2v13.5zM8 4.41c1.387-1.425 4.854 1.07 0 4.277C3.146 5.48 6.613 2.986 8 4.412z"/>
",
    }
  }
  fill="yellow"
  height={24}
  style={
    {
      "verticalAlign": "-0.125em",
    }
  }
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`QuestionIcon renders correctly 1`] = `
<svg
  dangerouslySetInnerHTML={
    {
      "__html": "
    <path d="M5 8a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm4 0a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm3 1a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"/>
    <path d="m2.165 15.803.02-.004c1.83-.363 2.948-.842 3.468-1.105A9.06 9.06 0 0 0 8 15c4.418 0 8-3.134 8-7s-3.582-7-8-7-8 3.134-8 7c0 1.76.743 3.37 1.97 4.6a10.437 10.437 0 0 1-.524 2.318l-.003.011a10.722 10.722 0 0 1-.244.637c-.079.186.074.394.273.362a21.673 21.673 0 0 0 .693-.125zm.8-3.108a1 1 0 0 0-.287-.801C1.618 10.83 1 9.468 1 8c0-3.192 3.004-6 7-6s7 2.808 7 6c0 3.193-3.004 6-7 6a8.06 8.06 0 0 1-2.088-.272 1 1 0 0 0-.711.074c-.387.196-1.24.57-2.634.893a10.97 10.97 0 0 0 .398-2z"/>
    ",
    }
  }
  fill="green"
  height={24}
  style={
    {
      "verticalAlign": "-0.125em",
    }
  }
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`SearchIcon renders correctly 1`] = `
<svg
  dangerouslySetInnerHTML={
    {
      "__html": "
      <path d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"/>
      ",
    }
  }
  fill="red"
  height={20}
  style={
    {
      "verticalAlign": "-0.125em",
    }
  }
  viewBox="0 0 16 16"
  width={20}
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`UpvoteIcon renders correctly 1`] = `
<svg
  dangerouslySetInnerHTML={
    {
      "__html": "
<path d="M2 16a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H2zm6.5-4.5V5.707l2.146 2.147a.5.5 0 0 0 .708-.708l-3-3a.5.5 0 0 0-.708 0l-3 3a.5.5 0 1 0 .708.708L7.5 5.707V11.5a.5.5 0 0 0 1 0z"/>",
    }
  }
  fill="green"
  height={24}
  style={
    {
      "verticalAlign": "-0.125em",
    }
  }
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
/>
`;
